doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -926,14 +909,257 @@ the conversion in hypotheses :n:`{+ @ident}`.
+      correspond to constants to be expanded before normal transparent
+      constants, while positive levels correspond to constants to be
+      expanded after normal transparent constants.
+    + ``expand`` : level of constants that should be expanded first (behaves
+      like −∞)
+    + ``transparent`` : Equivalent to level 0
+
+.. cmd:: Print Strategy @reference
+
+   This command prints the strategy currently associated with :n:`@reference`. It
+   fails if :n:`@reference` is not an unfoldable reference, that is, neither a
+   variable nor a constant.
+
+   .. exn:: The reference is not unfoldable.
+      :undocumented:
+
+.. cmd:: Print Strategies
+
+   Print all the currently non-transparent strategies.
+
+.. cmd:: Declare Reduction @ident := @red_expr
+
+   Declares a short name for the reduction expression :n:`@red_expr`, for
+   instance ``lazy beta delta [foo bar]``. This short name can then be used
+   in :n:`Eval @ident in` or ``eval`` constructs. This command
+   accepts the :attr:`local` attribute, which indicates that the reduction
+   will be discarded at the end of the
+   file or module. The name is not qualified. In
+   particular declaring the same name in several modules or in several
+   functor applications will be rejected if these declarations are not
+   local. The name :n:`@ident` cannot be used directly as an Ltac tactic, but
+   nothing prevents the user from also performing a
+   :n:`Ltac @ident := @red_expr`.
+
+   .. seealso:: :ref:`applyingconversionrules`
+
+.. todo PR
+   move rewriting.rst to equality.rst once the text is final,
+   perhaps in a different PR to keep the history somewhat understandable
+
+Equality
+--------
+
+.. tacn:: f_equal
+
+   For a goal with the form :n:`f a__1 ... a__n = g b__1 ... b__n`, creates
+   subgoals :n:`f = g` and :n:`a__i = b__i` for the `n` arguments. SImple subgoals, such as those
+   provable by :tacn:`reflexivity` or :tacn:`congruence` are solved automatically.
+
+.. tacn:: reflexivity
+
+   For a goal with the form `t=u`, verifies that `t` and `u` are :term:`convertible`,
+   and if so,
+   solves the goal.  If not, it fails.  It is equivalent to ``apply refl_equal``.
+
+   .. todo PR: this doesn't cover how reflexibity can solve the goal "True"
+
+      also for te goal "1 + 2 = 4", this generates "Unable to unify ..." with a stack
+      trace.  Should have a better message, right?
+
+   .. exn:: The conclusion is not a substitutive equation.
+      :undocumented:
+
+   .. exn:: Unable to unify ... with ...
+      :undocumented:
+
+.. tacn:: symmetry {? in @goal_occurrences }
+
+   Changes a goal that has the form :g:`t = u` into :g:`u = t`.  :n:`@goal_occurrences`
+   may be used to make the change in the selected hypotheses and/or the conclusion.
+   The `at` clause of :n:`@goal_occurrences` is not supported.
+
+.. tacn:: transitivity @one_term
+
+   Changes a goal that has the form :g:`t=u`
+   into the two subgoals :n:`t=@one_term` and :n:`@one_term=u`.
```

Comment:
Note that, like `reflexivity`, `symmetry` and `transitivity` and `etransitivity` will also run `intros` first and will also perform typeclass search.  Furthermore, all of these tactics also seem built to work on any inductive family with a single constructor whose type matches any of the types in the comments at https://github.com/coq/coq/blob/16765871394a81975047b37f15a902fcc112dc40/tactics/hipattern.ml#L277-L289.  However, the handling of `symmetry` seems to have a bug, and I've just reported https://github.com/coq/coq/issues/13838